### PR TITLE
added PSQP negative informs

### LINKED ISSUE #146: Missing informs value for PyPSQP
## Type of issue
What types of issue is it?
_Select the appropriate type(s) that describe this issue_

- Bugfix (non-breaking change which fixes an issue)

## Description
The wrapper for PSQP defines mappings from return values to strings

```
self.informs = {
--
1: "Change in design variable was less than or equal to tolerance",
2: "Change in objective function was less than or equal to tolerance",
3: "Objective function less than or equal to tolerance",
4: "Maximum constraint value is less than or equal to tolerance",
11: "Maximum number of iterations exceeded",
12: "Maximum number of function evaluations exceeded",
13: "Maximum number of gradient evaluations exceeded",
-6: "Termination criterion not satisfied, but obtained point is acceptable",
}

```

But PSQP can have a return code of -10

```
     IF (IREST.GT.0) THEN
         CALL MXDSMI(N,H)
         LD = MIN(LD,1)
         IDECF = 1
         IF (KIT.LT.NIT) THEN
             NRES = NRES + 1
             KIT = NIT
         ELSE
             ITERM = -10
             IF (ITERS.LT.0) ITERM = ITERS - 5
             GO TO 80
         END IF
     END IF
```

So I get exceptions for that case. 

```
  File “/home/travis/miniconda/envs/PY3.8/lib/python3.8/site-packages/pyoptsparse/pyPSQP/pyPSQP.py”, line 261, in __call__
    sol_inform[“text”] = self.informs[inform]
KeyError: -10
```
I would submit a PR but I don't know what text to use to describe this termination condition.

### PR DESCRIPTION
**Pull request by sseraj**
## Purpose
Fixes #146 by adding some negative informs and redirecting unspecified negative values to `-10`.

## Type of change
- Bugfix (non-breaking change which fixes an issue)

## Checklist
_Put an `x` in the boxes that apply._

- [x] I have run `flake8` and `black` to make sure the code adheres to PEP-8 and is consistently formatted
- [x] I have run unit and regression tests which pass locally with my changes
- [ ] I have added new tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation

### CODE REVIEW

**Comment by sseraj** [comment]
@hschilling Please check that this resolves the issue

File: pyoptsparse/pyPSQP/pyPSQP.py [hunk 1]
```diff
@@ -61,6 +61,9 @@ def __init__(self, raiseError=True, *args, **kwargs):
             12: "Maximum number of function evaluations exceeded",
             13: "Maximum number of gradient evaluations exceeded",
             -6: "Termination criterion not satisfied, but obtained point is acceptable",
+            -7: "Positive directional derivative in line search",
+            -8: "Error in interpolation in line search",
```
Comment (nwu63) [inline pyoptsparse/pyPSQP/pyPSQP.py:65]: Could you rephrase this as `Interpolation error in line search`?

**Comment by hschilling** [comment]
@nwu63 and @sseraj Thank you! That fixed our problem. That test and all other tests that use pyoptsparse pass

**Comment by nwu63** [comment]
Great, I'll approve and merge.

**Comment by hschilling** [comment]
Just asking. When will there be a release with this PR in it? Thanks

**Comment by nwu63** [comment]
We've been making releases on an ad-hoc bases -- there is no set schedule or anything. In the past I've been waiting until there's a couple of bug fixes before making a release, but if this is something you guys want (I assume the OpenMDAO team is testing against specific releases of pyOptSparse) I can certainly make a release asap.

**Comment by hschilling** [comment]
Completely understand that you would rather group bug fixes into releases. But we do test against specific releases, so if it isn't too much trouble, a release for this would be great. Thanks for your incredibly quick responses.

**Comment by nwu63** [comment]
Not a problem. In the future if you want certain bug fixes to be made into a release just let us know, I understand that you want those tests to pass and you don't want to test against master. I've just released v2.1.5 for pyOptSparse.

**Comment by hschilling** [comment]
Thanks !
